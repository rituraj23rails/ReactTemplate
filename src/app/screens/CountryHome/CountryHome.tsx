import React, { Component, useRef } from 'react';
import './CountryHome.css';
import WithRouter from '../../../WithRouter';

export interface Props {
    navigate: any;
    id: string;
}

interface S {
    countryName: any;
    countryList: any;
}

interface SS {
    id: any;
}

class CountryHome extends Component<Props, S, SS> {
    carousel: any = null;
    constructor(props: any) {
        super(props);
        this.state = {
            countryName: '',
            countryList: []
        };
    };

    componentDidMount = async () => {
    }

    componentWillUnmount = () => {

    }

    handleChangeTextInput = (event: any) => {
        event?.preventDefault();
        this.setState({ countryName: event.target.value });
    }

    onSubmitButton = async () => {
        try {
            let response = await fetch(`https://restcountries.com/v3.1/name/${this.state.countryName}`);
            let json = await response.json();
            this.setState({ countryList: json }, () => {
                console.log("@@@ Country List Data ==============", this.state.countryList)
                if (json.status === 404) {
                    alert('No Data found.');
                    return;
                }
                this.props.navigate(`/countrylist`, { state: { countryList: this.state.countryList }});
            });
        } catch (error) {
            alert(error)
        }
    }

    renderInputCountryForm = () => {
        return (
            <div className="formBox">
                <p className="headingText">Welcome To Country App</p>
                <input data-testid="textInputTestID" type="text" className='textInputField' value={this.state.countryName} onChange={(event) => this.handleChangeTextInput(event)} />
                <button data-testid="submitButtonTestID" onClick={() => this.onSubmitButton()} className="submitButton">SUBMIT</button>
            </div>
        )
    }

    render() {
        return (
            <div className='mainContainer'>
                {this.renderInputCountryForm()}
            </div>
        )
    }
};

export default WithRouter(CountryHome);
export {CountryHome};
