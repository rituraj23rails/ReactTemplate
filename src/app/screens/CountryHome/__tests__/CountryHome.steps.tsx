import { defineFeature, loadFeature } from "jest-cucumber"
import { shallow, ShallowWrapper, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { CountryHome } from "../CountryHome";

configure({adapter: new Adapter()});

const screenProps = {
    navigate: jest.fn(),
    id: "CountryHome"
}

const feature = loadFeature('./src/app/screens/CountryHome/__tests__/features/CountryHome-scenarios.feature');

defineFeature(feature, (test) => {

    test('User navigates Country Home Screen', ({ given, when, then }) => {
        let CountryHomeScreenWrapper: ShallowWrapper;
        let CountryHomeScreenInstance: CountryHome;

        given('I am a User loading Country Home Screen', () => {
            CountryHomeScreenWrapper = shallow(<CountryHome {...screenProps} />)
        });

        when('I navigate to the Country Home Screen', () => {
            CountryHomeScreenInstance = CountryHomeScreenWrapper.instance() as CountryHome
        });

        then('Country Home Screen will load with out errors', () => {
            CountryHomeScreenInstance.componentDidMount();
            expect(CountryHomeScreenWrapper).toBeTruthy()
            expect(CountryHomeScreenWrapper).toMatchSnapshot()
        });

        then('I can enter search input value for countries', () => {
            let inputComponent = CountryHomeScreenWrapper.findWhere((node) => node.prop("data-testid") === "textInputTestID");
            inputComponent.simulate('change', { preventDefault: jest.fn(), target: { value: 'ind' } });
        });
        
        then('I can select the submit button to search the countries', () => {
            let buttonComponent = CountryHomeScreenWrapper.findWhere((node) => node.prop("data-testid") === "submitButtonTestID");
            buttonComponent.simulate("click");
        });

        then('I can leave the screen with out errors', () => {
            CountryHomeScreenInstance.componentWillUnmount();
            expect(CountryHomeScreenWrapper).toBeTruthy()
            expect(CountryHomeScreenWrapper).toMatchSnapshot()
        });
    });

});
