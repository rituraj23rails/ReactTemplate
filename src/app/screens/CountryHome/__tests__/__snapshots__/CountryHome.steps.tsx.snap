// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country Home Screen User navigates Country Home Screen 1`] = `
<div
  className="mainContainer"
>
  <div
    className="formBox"
  >
    <p
      className="headingText"
    >
      Welcome To Country App
    </p>
    <input
      className="textInputField"
      data-testid="textInputTestID"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="submitButton"
      data-testid="submitButtonTestID"
      onClick={[Function]}
    >
      SUBMIT
    </button>
  </div>
</div>
`;

exports[`Country Home Screen User navigates Country Home Screen 2`] = `
<div
  className="mainContainer"
>
  <div
    className="formBox"
  >
    <p
      className="headingText"
    >
      Welcome To Country App
    </p>
    <input
      className="textInputField"
      data-testid="textInputTestID"
      onChange={[Function]}
      type="text"
      value="ind"
    />
    <button
      className="submitButton"
      data-testid="submitButtonTestID"
      onClick={[Function]}
    >
      SUBMIT
    </button>
  </div>
</div>
`;
